doc/decisions/map_lazyloading.md

```diff
@@ -0,0 +1,58 @@
+# Map Lazyloading
+
+## Problem
+
+The application's main component is the map, which consists of multiple layers and even more objects.
+One more thing to consider is that the map consists not only of the data for the current date, but also for the past and future dates e.g. harvest time, planting history.
+However, not all layers are visible simultaneously, nor are they required during the initial loading of the application.
+
+Loading all the data at once during the initial load may result in slower performance, therefore a strategy must be developed to lazyload data on the map to optimize its performance.
+In other words, the data which is not required for the user to start working with the map should not be loaded during the startup.
+
+## Constraints
+
+N/A
+
+## Assumptions
+
+1. The majority of layers are only modified once a year
+2. The loading of all the data at once during the initial load will result in slower performance
+
+## Solutions
+
+### Full fetch
+
+Fetching all the data from the backend during the initial load, while making the map available immediately, may not be the most optimal solution as it would lead to a slower initial load time.
```

Comment:
I kindly disagree here, since there was no requirement defined in the project to prove every possible solution. Of course, it is possible to measure the load time for this case, which I actually did in my study. However, this case is obvious in my opinion that the partially loaded data provides better performance than the full load. Please correct me if I'm wrong. The main idea of the paragraph "Full fetch" was to provide a small comment that such an idea was once considered.

I see 2 possible improvements here:
- rename the title of the PR, which could be confusing, from "Feasibility study" to "Architecture decisions of the canvas". Because these decisions are the outcomes and not the actual study
- add an assumption that the loading of the full structure is slower than the partial one.

As I've said I think the assumption is unnecessary and such measurements on each idea will just steal precious `PermaplanT` time

Reply:
What I wanted to say: if we have data, it should be written in decisions. That partial loads are faster than full loads is obvious, but what is not clear to the reader if a full fetch might be feasible or to which extend it might be. If we have data to which extend it works, it makes the decision much more interesting to read.